hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -262,57 +336,34 @@ public void onCompactionCompleted(
               }
-              // Let us also build the graph
-              for (String outFilePath : compactionJobInfo.outputFiles()) {
-                String outfile =
-                    Paths.get(outFilePath).getFileName().toString();
-                CompactionNode outfileNode = compactionNodeTable.get(outfile);
-                if (outfileNode == null) {
-                  long numKeys = 0;
-                  try {
-                    numKeys = getSSTFileSummary(outfile);
-                  } catch (Exception e) {
-                    LOG.warn(e.getMessage());
-                  }
-                  outfileNode = new CompactionNode(outfile,
-                      lastSnapshotPrefix, numKeys,
-                      currentCompactionGen);
-                  compactionDAGFwd.addNode(outfileNode);
-                  compactionDAGReverse.addNode(outfileNode);
-                  compactionNodeTable.put(outfile, outfileNode);
-                }
-                for (String inFilePath : compactionJobInfo.inputFiles()) {
-                  String infile =
-                      Paths.get(inFilePath).getFileName().toString();
-                  CompactionNode infileNode = compactionNodeTable.get(infile);
-                  if (infileNode == null) {
-                    long numKeys = 0;
-                    try {
-                      numKeys = getSSTFileSummary(infile);
-                    } catch (Exception e) {
-                      LOG.warn(e.getMessage());
-                    }
-                    infileNode = new CompactionNode(infile,
-                        lastSnapshotPrefix,
-                        numKeys, UNKNOWN_COMPACTION_GEN);
-                    compactionDAGFwd.addNode(infileNode);
-                    compactionDAGReverse.addNode(infileNode);
-                    compactionNodeTable.put(infile, infileNode);
-                  }
-                  if (outfileNode.fileName.compareToIgnoreCase(
-                      infileNode.fileName) != 0) {
-                    compactionDAGFwd.putEdge(outfileNode, infileNode);
-                    compactionDAGReverse.putEdge(infileNode, outfileNode);
-                  }
-                }
-              }
-              if (debugEnabled(DEBUG_DAG_BUILD_UP)) {
-                printMutableGraph(null, null, compactionDAGFwd);
+              sb.append('\n');
+
+              // Persist infile/outfile to file
+              try (BufferedWriter bw = Files.newBufferedWriter(
+                  Paths.get(CURRENT_COMPACTION_LOG_FILENAME),
+                  StandardOpenOption.CREATE, StandardOpenOption.APPEND)) {
+                bw.write(sb.toString());
+                bw.flush();
+              } catch (IOException e) {
+                throw new RuntimeException(
+                    "Failed to append to log for DB compaction tracking", e);
               }
+
+              // Let us also build the graph
+//              populateCompactionDAG(compactionJobInfo.inputFiles(),
```

Comment:
I'm not sure if this comment means that populateCompactionDAG() is going to be removed here, but it doesn't seem like it should be removed.

(since it creates the compactionNodes and their snapshotGeneration field is set to the lastSnapshotCounter at the time of creation.)

Reply:
I have restored the commented out lines in https://github.com/apache/ozone/pull/3824 . This was commented out to test the DAG reconstruction (read from compaction log) in the UT.